### DISCUSSION TITLE
Adapt TypeScript to PouchDB

### DISCUSSION THREAD

**Discussion by mqtik** [pouchdb/pouchdb#8501] [General]
I think PouchDB codebase would benefit not only from ES6 features, but also TypeScript.

We can't be commenting everything in the code, it should be self explanatory, and it would be extremely helpful for new contributors to know what params a certain function expects, with all the types laid out.

I'm happy to help with this.

**Comment by karlwestin** [comment]
hi!
you opened a similar issue last year https://github.com/pouchdb/pouchdb/issues/8307

there is a pouchdb types thing to import https://www.npmjs.com/package/@types/pouchdb
unfortunately the typescript users don't seem very keen on keeping it updated
we often hear the argument "it would be much more beneficial if the whole library was written in typescript", but the problem is that the toolchain is already complex and doing that would raise the maintenance burden - please understand that not everyone is set up with typescript and not everyone likes it. 

I think it's a better choice to update the type definitions and work with those.

**Reply by louwers** [reply]
JSDoc is an excellent choice if you don't want to bother with additional tooling.
